### PR TITLE
Change parity test structure

### PR DESCRIPTION
**Pull request by dakingha69**
This adds `yarn` scripts for running the parity tests from project root. Before it was not possible to easily run the test from the initial project location. Now you can run
```bash
yarn test-parity-native
# OR
yarn test-parity-docker
```
to run the tests.

The test file structure is also streamlined through this PR. Also some the tests were refactored regarding code style.

### CODE REVIEW

**Comment by dakingha69** [comment]
Also when running the native tests, the ones testing the connecting peers fail... Any idea why?

File: scripts/run_native_parity_tests.sh [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
 
 echo "Starting node #0 ..."
 source ./setup_node.sh --test &> /dev/null
 
 echo "Starting node #1 ..."
 export NODE_ID=1
-source  ./setup_node.sh --test &> /dev/null
+source ./setup_node.sh --test &> /dev/null
+
+sleep 5
```
Comment (paguos) [inline scripts/run_native_parity_tests.sh]: We should never use this (sleep is not deterministic) ... The setup processes can be longer on different computers. The correct approach is to retry. `mocha` has good support there. That is the reason you're getting also some failures in your tests.  

I would keep to this approach... 

```js 
      // delay in each retry after first failure
      beforeEach(function(done) {
        if (this.currentTest.currentRetry() > 0) {
          setTimeout(done, this.currentTest.currentRetry() * 500);
        } else {
          done();
        }
      });
      this.retries(10);
```

File: test/parity-authority/docker.test.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+const assert = require("assert");
+const request = require("sync-request");
```
Comment (paguos) [inline test/parity-authority/docker.test.js]: You're using `await` so we don't need `sync-request` anymore, we should change it to the standard `request` 😉

File: scripts/run_native_parity_tests.sh (hunk 1 above)
Comment (dakingha69) [reply scripts/run_native_parity_tests.sh]: Ah ok yes makes sense. Readded it. It just looks a bit hacky :D But I can still not get the peer count tests running for native

File: test/parity-authority/docker.test.js (hunk 2 above)
Comment (dakingha69) [reply test/parity-authority/docker.test.js]: ok done

File: scripts/run_native_parity_tests.sh (hunk 1 above)
Comment (paguos) [reply scripts/run_native_parity_tests.sh]: It works for me ... Somehow the `bootnode` must be falling 😞

**Pull request merged**